### PR TITLE
[MDS-5196] - Common package Typescript setup

### PR DESCRIPTION
**Pull request by matbusby-fw**
## Objective 

Update configuration in the common workspace to properly handle the export of TS files and types.
PoC using Notice of Departures with interface imports from the common package.

[MDS-5196](https://bcmines.atlassian.net/browse/MDS-5196)

### CODE REVIEW

File: services/minespace-web/src/components/dashboard/mine/noticeOfDeparture/NoticeOfDeparture.tsx [hunk 1]
```diff
@@ -23,32 +23,30 @@ import {
...
+}
+
+export const NoticeOfDeparture: FC<NoticeOfDepartureProps> = (props) => {
   const { mine, nods, permits } = props;
   const [isLoaded, setIsLoaded] = useState(false);
   const location = useLocation();
   const url = useParams();
 
   const handleFetchPermits = async () => {
+    console.log(mine.mine_guid);
```
Comment (taraepp) [inline services/minespace-web/src/components/dashboard/mine/noticeOfDeparture/NoticeOfDeparture.tsx]: stray console log!

**Review by taraepp — APPROVED** [review]
LGTM

File: services/common/package.json [hunk 2]
```diff
@@ -38,6 +39,7 @@
     "react-hot-loader": "4.12.21",
     "style-loader": "0.23.1",
     "terser-webpack-plugin": "1.3.0",
+    "typescript": "^5.0.4",
```
Comment (simensma-fresh) [inline services/common/package.json:42]: Want to lock this version down?

File: services/common/src/interfaces/amendment.interface.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+export interface IAmendment {
+  permit_amendment_guid: string;
+  permit_amendment_status_code: string;
+  permit_amendment_type_code: string;
+  permit_ammendment_id: string;
```
Comment (simensma-fresh) [inline services/common/src/interfaces/amendment.interface.ts:5]: Extra `m`? 😅

**Review by simensma-fresh — APPROVED** [review]
Just a couple of minor comments, otherwise looks good. Excited to test this out! Do you think it's good to be merged in? or should we hold off until after the next release in case something blows up?

File: services/common/src/interfaces/amendment.interface.ts (hunk 3 above)
Comment (matbusby-fw) [reply services/common/src/interfaces/amendment.interface.ts:5]: Ugh, didn't see your comments until after I had merged.  I'll do a quick PR to fix these things and keep an eye on things.  I'm pretty sure the likelyhood of it blowing up is small, but perhaps I'll just monitor for now.

Comment (simensma-fresh) [reply services/common/src/interfaces/amendment.interface.ts:5]: Totally up to you! Monitoring sounds good. It's always easy enough to do a revert if needed

**Pull request merged**